crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1321,34 +1352,140 @@ impl<'src> Lexer<'src> {
 
     /// Restore the lexer to the given checkpoint.
     pub(crate) fn rewind(&mut self, checkpoint: LexerCheckpoint<'src>) {
-        self.current_value = checkpoint.value;
-        self.current_kind = checkpoint.current_kind;
-        self.current_range = checkpoint.current_range;
-        self.cursor = checkpoint.cursor;
-        self.state = checkpoint.state;
-        self.nesting = checkpoint.nesting;
-        self.indentations.rewind(checkpoint.indentations_checkpoint);
-        self.pending_indentation = checkpoint.pending_indentation;
-        self.fstrings.rewind(checkpoint.fstrings_checkpoint);
-        self.errors.truncate(checkpoint.errors_position);
+        let LexerCheckpoint {
+            value,
+            current_kind,
+            current_range,
+            current_flags,
+            cursor,
+            state,
+            nesting,
+            indentations_checkpoint,
+            pending_indentation,
+            fstrings_checkpoint,
+            errors_position,
+        } = checkpoint;
+
+        self.current_value = value;
+        self.current_kind = current_kind;
+        self.current_range = current_range;
+        self.current_flags = current_flags;
+        self.cursor = cursor;
+        self.state = state;
+        self.nesting = nesting;
+        self.indentations.rewind(indentations_checkpoint);
+        self.pending_indentation = pending_indentation;
+        self.fstrings.rewind(fstrings_checkpoint);
+        self.errors.truncate(errors_position);
     }
 
     pub fn finish(self) -> Vec<LexicalError> {
         self.errors
     }
 }
 
+// Another way to represent token flags
+pub(crate) enum TokenFlags2 {
+    String(AnyStringFlags),
+    FStringStart(AnyStringFlags),
+    Other,
+}
+
+bitflags! {
+    #[derive(Clone, Copy, Debug)]
+    pub(crate) struct TokenFlags: u8 {
```

Comment:
I might be missing something, but it looks like this is identical to the `AnyStringFlags` type we already have in `nodes.rs` in terms of its interface and capabilities. Couldn't we just reuse that type? It was originally designed for use in the lexer

Reply:
Yes, currently it is equivalent to the `AnyStringFlags` but we do plan to add additional flags which might not be related to a string.

Reply:
Got it, thanks!